File: packages/argus/tests/fixtures/sample.kt
package com.example.app

import foo.Bar as Baz
import foo.*
import kotlin.collections.List

interface Repo {
  fun load(): String
}

object Singleton {
  val count: Int = 0
  fun bump() {}
}

enum class Status { ACTIVE, INACTIVE }

typealias Handler = (String) -> Unit

class A : Base(), Repo, Serializable {
  val id: Int = 1
  var name: String = ""

  constructor(x: Int) : this() { }

  fun f() {}

  operator fun plus(other: A): A = other

  companion object Factory {
    fun create(): A = A()
  }
}

fun String.extension() {}

val topLevel = 1

File: packages/argus/tests/fixtures/sample.dart
import 'dart:io';
part 'sample_part.dart';

typedef IntCallback = void Function(int value);

const double kPadding = 8.0;
final config = AppConfig();

enum Status { active, inactive }

mixin Disposable on Base {
  void dispose() {}
}

class A extends B with Disposable implements C {
  final String name;
  int count = 0;

  A(this.name);
  A.named({required this.name});
  factory A.create() => A('default');

  void m() {}
  void build() {
    final list = <int>[];
    list..add(1)..remove(2);
  }
}
